### PR TITLE
Add Locale Switching

### PR DESCRIPTION
**Pull request by Abban**
This adds backend functionality for swapping locales and a new soft dependency in the FunFunFactory

Ticket: https://phabricator.wikimedia.org/T256612

### CODE REVIEW

File: app/EventHandlers/StoreLocale.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		$request->setLocale( $locale );
+		$this->factory->setLocale( $locale );
+	}
+
+	private function findLocale( Request $request ): string {
+		if ( $request->cookies->has( 'locale' ) ) {
+			return $request->cookies->get( 'locale' );
+		}
+
+		return $request->getPreferredLanguage();
```
Comment (gbirke) [inline app/EventHandlers/StoreLocale.php:48]:
```suggestion
		return $request->getPreferredLanguage() ?? '';
```

**Review by gbirke — CHANGES_REQUESTED** [review]
While we saw that `StoreLocale` was implicitly tested in our E2E test, please create a separate unit test that tests the fallback mechanism.

File: src/Factories/FunFunFactory.php [hunk 2]
```diff
@@ -504,7 +502,7 @@ private function getDefaultTwigVariables(): array {
 		return [
 			'honorifics' => $this->getHonorifics()->getList(),
 			'piwik' => $this->config['piwik'],
-			'locale' => $this->config['locale'],
+			'locale' => $this->getLocale(),
```
Comment (gbirke) [inline src/Factories/FunFunFactory.php:505]: You can drop this, locale can now be determined on the client side.

File: src/Factories/FunFunFactory.php [hunk 3]
```diff
@@ -772,7 +770,7 @@ public function getInternalErrorHtmlPresenter(): ExceptionHtmlPresenterInterface
 			'Error_Page.html.twig',
 			[
 				'piwik' => $this->config['piwik'],
-				'locale' => $this->config['locale'],
+				'locale' => $this->getLocale(),
```
Comment (gbirke) [inline src/Factories/FunFunFactory.php:773]: You can drop this, locale can now be determined on the client side.

**Review by gbirke — APPROVED** [review]

**Comment by gbirke** [comment]
For a successful deploy, please make a PR in fundraising-infrastructure that removes the `locale` key from the configs

**Comment by Abban** [comment]
> For a successful deploy, please make a PR in fundraising-infrastructure that removes the `locale` key from the configs

Locale is defaulted by the dist config and not overridden by the ones in infrastructure AFAICT

**Pull request merged**